AnimationSystem/AnimationSystem.cc

```diff
@@ -0,0 +1,1074 @@
+          else if (boneName == "Left_indexFinger3") boneIndexes.Left_indexFinger3 = i;
+          else if (boneName == "Left_ringFinger1") boneIndexes.Left_ringFinger1 = i;
+          else if (boneName == "Left_ringFinger2") boneIndexes.Left_ringFinger2 = i;
+          else if (boneName == "Left_ringFinger3") boneIndexes.Left_ringFinger3 = i;
+          else if (boneName == "Left_littleFinger1") boneIndexes.Left_littleFinger1 = i;
+          else if (boneName == "Left_littleFinger2") boneIndexes.Left_littleFinger2 = i;
+          else if (boneName == "Left_littleFinger3") boneIndexes.Left_littleFinger3 = i;
+          else if (boneName == "Right_shoulder") boneIndexes.Right_shoulder = i;
+          else if (boneName == "Right_arm") boneIndexes.Right_arm = i;
+          else if (boneName == "Right_elbow") boneIndexes.Right_elbow = i;
+          else if (boneName == "Right_wrist") boneIndexes.Right_wrist = i;
+          else if (boneName == "Right_middleFinger1") boneIndexes.Right_middleFinger1 = i;
+          else if (boneName == "Right_middleFinger2") boneIndexes.Right_middleFinger2 = i;
+          else if (boneName == "Right_middleFinger3") boneIndexes.Right_middleFinger3 = i;
+          else if (boneName == "Right_thumb0") boneIndexes.Right_thumb0 = i;
+          else if (boneName == "Right_thumb1") boneIndexes.Right_thumb1 = i;
+          else if (boneName == "Right_thumb2") boneIndexes.Right_thumb2 = i;
+          else if (boneName == "Right_indexFinger1") boneIndexes.Right_indexFinger1 = i;
+          else if (boneName == "Right_indexFinger2") boneIndexes.Right_indexFinger2 = i;
+          else if (boneName == "Right_indexFinger3") boneIndexes.Right_indexFinger3 = i;
+          else if (boneName == "Right_ringFinger1") boneIndexes.Right_ringFinger1 = i;
+          else if (boneName == "Right_ringFinger2") boneIndexes.Right_ringFinger2 = i;
+          else if (boneName == "Right_ringFinger3") boneIndexes.Right_ringFinger3 = i;
+          else if (boneName == "Right_littleFinger1") boneIndexes.Right_littleFinger1 = i;
+          else if (boneName == "Right_littleFinger2") boneIndexes.Right_littleFinger2 = i;
+          else if (boneName == "Right_littleFinger3") boneIndexes.Right_littleFinger3 = i;
+          else if (boneName == "Right_leg") boneIndexes.Right_leg = i;
+          else if (boneName == "Right_knee") boneIndexes.Right_knee = i;
+          else if (boneName == "Right_ankle") boneIndexes.Right_ankle = i;
+          else if (boneName == "Right_toe") boneIndexes.Right_toe = i;
+          else if (boneName == "Left_leg") boneIndexes.Left_leg = i;
+          else if (boneName == "Left_knee") boneIndexes.Left_knee = i;
+          else if (boneName == "Left_ankle") boneIndexes.Left_ankle = i;
+          else if (boneName == "Left_toe") boneIndexes.Left_toe = i;
+        }
+      }
+
+      // -------------------------------------------------------------------------
+
+      jsonStr += "[";
```

Comment:
Instead of writing your own JSON, I think there are better JSON libraries to that we can import that are less likely to error in corner cases, link non-utf8 data.

Reply:
I've successfully used https://github.com/nlohmann/json to parse JSON strings from JS side.
But I can't find `stringify` feature in it.
And very surprisingly that, because of type and reflection reason, it's very hard to find a JSON stringify C++ lib.
I'll keep investigating.

Reply:
Trying to use this lib
https://cppsecrets.com/users/13187108971001001049711497106484964103109971051084699111109/C00-PocoJSONObjectstringify.php
https://github.com/pocoproject
https://github.com/pocoproject/poco

Reply:
After tried many cpp json libs and failed to integrate ( mainly because of Undefined Symbol error ).
I investigated "nlohmann/json" again ( it's a single header cpp lib, very easy to integrate ).
And found it really has `stringify()` feature, but named `bump()` instead.
https://github.com/webaverse/app-wasm/compare/19d0d52831bc4ca52d7e768dc125bf477bbbb24e...29630e753c58e9aceb7a6fb0e09532f414cada20